elastic/kibana
# [Form lib] Fix race condition when running validation

This PR fixes an issue found by @Dosant while working on https://github.com/elastic/kibana/pull/138748.

In some edge cases there was a race conditions that cleared a timeout in the form lib responsible for delaying the error message rendered in the UI (500ms between the last key stroke in the field and the moment the error is displayed under the field).

As the timeout was cleared, this caused the error to never be displayed.

Note that this issue was only a UI rendering issue, underneath the form was correctly invalid and the user could not send the form.

**Note:** I tried to add a component integration test to prevent regression on the form lib for this behaviour with no luck. It seems that the order of execution of `useEffect()` inside `await act()` calls differs from the execution order in the browser. I could not reproduce the race condition within the test.

## How to test

* Revert the change in this PR
* Navigate to the stack management > ILM policies app
* Create a policy and activate "warm" and "cold" phases
* Under "_Move data into phase when_" set a **lower** value in the cold phase (e.g. 2) than in the warm (e.g. 4)
* The form is invalid but no error is displayed underneath the field (in the cold phase)
* Put back the change in this PR
* Do the above steps: now the error should be displayed underneath the field

<img width="1012" alt="Screenshot 2022-08-24 at 17 22 30" src="https://user-images.githubusercontent.com/2854616/186457959-a1a5ba54-f6d6-49ca-842b-77dae64114ed.png">
